src/utils.js

```
@@ -359,19 +371,24 @@ export const getLightBackground = (alpha) => {
 };
 
 /**
- * Render value to integer or decimal
+ * Render value in a formatted way for display
  *
- * @params {number} amount Amount to render
- * @params {boolean} isInteger If it's an integer or decimal
- *
- * @return {string} rendered value
+ * @param {bigint} amount The token amount as BigInt
+ * @param {boolean} isInteger Whether the token is an NFT or regular token
+ * @return {string} Formatted value for display
  */
 export const renderValue = (amount, isInteger) => {
-  if (isInteger) {
-    return hathorLib.numberUtils.prettyIntegerValue(amount);
-  }
+  try {
+    if (isInteger) {
+      return hathorLib.numberUtils.prettyIntegerValue(amount);
+    }
 
-  return hathorLib.numberUtils.prettyValue(amount);
+    return hathorLib.numberUtils.prettyValue(amount);
+  } catch (e) {
+    console.error('Failed to render value:', e);
+    // Return a fallback string in case of error
+    return '0';
```

Should we just let it crash?